x/concentrated-liquidity/math/precompute.go

```diff
@@ -17,8 +18,67 @@ var (
 	osmomathBigTenDec = osmomath.NewBigDec(10)
 	bigPowersOfTen    []osmomath.BigDec
 	bigNegPowersOfTen []osmomath.BigDec
+
+	// 9 * 10^(-types.ExponentAtPriceOne), where types.ExponentAtPriceOne is non-positive and is s.t.
+	// this answer fits well within an int64.
+	geometricExponentIncrementDistanceInTicks = 9 * sdk.NewDec(10).PowerMut(uint64(-types.ExponentAtPriceOne)).TruncateInt64()
 )
 
+// Builds metadata for every additive tickspacing exponent, namely:
+// * what is the first price this tick spacing applies to
+// * what is the first tick this applies for
+// * (saves on pre-compute) what is the additive increment per tick.
+//
+// This would be stored in a map, keyed by:
+// 0 => (1.00, 10^(types.ExponentAtPriceOne), 0)
+// 1 => (10, 10^(1 + types.ExponentAtPriceOne), 9 * types.ExponentAtPriceOne)
+// 2 => (100, 10^(2 + types.ExponentAtPriceOne), 9 * (types.ExponentAtPriceOne + 1))
+// -1 => (0.1, 10^(types.ExponentAtPriceOne - 1), 9 * (types.ExponentAtPriceOne - 1))
+type tickExpIndexData struct {
+	// if price < initialPrice, we are not in this exponent range.
+	initialPrice sdk.Dec
+	// if price >= maxPrice, we are not in this exponent range.
+	maxPrice sdk.Dec
+	// TODO: Change to normal Dec, if min spot price increases.
+	// additive increment per tick here.
+	additiveIncrementPerTick osmomath.BigDec
+	// the tick that corresponds to initial price
+	initialTick int64
+}
+
+var tickExpCache map[int64]*tickExpIndexData = make(map[int64]*tickExpIndexData)
+
+func buildTickExpCache() {
+	// build positive indices first
+	maxPrice := sdkOneDec
+	curExpIndex := int64(0)
+	for maxPrice.LT(types.MaxSpotPrice) {
+		tickExpCache[curExpIndex] = &tickExpIndexData{
+			// price range 10^curExpIndex to 10^(curExpIndex + 1). (10, 100)
+			initialPrice:             sdkOneDec.Mul(sdkTenDec.Power(uint64(curExpIndex))),
+			maxPrice:                 sdkOneDec.Mul(sdkTenDec.Power(uint64(curExpIndex + 1))),
```

Comment:
Why do we multiply 1 here?

Reply:
Also, any reason for not using `PowTenInternal` or `powTenBigDec` here?

Reply:
oh true lol